dockstore-webservice/src/main/java/io/dockstore/webservice/core/Version.java

```diff
@@ -161,37 +147,46 @@
     public Version() {
         sourceFiles = new TreeSet<>();
         validations = new TreeSet<>();
-        doiStatus = DOIStatus.NOT_REQUESTED;
+        versionMetadata.doiStatus = DOIStatus.NOT_REQUESTED;
+        versionMetadata.parent = this;
     }
 
+    @ApiModelProperty(value = "Whether this version has been verified or not", position = 8)
     public boolean isVerified() {
-        return verified;
+        return getVersionMetadata().verified;
     }
 
     public void setVerified(boolean verified) {
-        this.verified = verified;
+        this.getVersionMetadata().verified = verified;
     }
 
+    @ApiModelProperty(value = "Verified source for the version", position = 9)
     public String getVerifiedSource() {
-        return verifiedSource;
+        return versionMetadata.verifiedSource;
     }
 
     public void setVerifiedSource(String verifiedSource) {
-        this.verifiedSource = verifiedSource;
+        this.getVersionMetadata().verifiedSource = verifiedSource;
     }
 
     public boolean isDirtyBit() {
         return dirtyBit;
     }
 
     public void setDirtyBit(boolean dirtyBit) {
-        this.dirtyBit = dirtyBit;
+        if (!this.isFrozen()) {
+            this.dirtyBit = dirtyBit;
+        }
     }
 
     void updateByUser(final Version version) {
-        reference = version.reference;
-        hidden = version.hidden;
-        this.setFrozen(version.frozen);
+        this.getVersionMetadata().hidden = version.isHidden();
```

Comment:
nb: So, the Dockstore User will be able to hide a snapshotted version.

Reply:
I think so, this might be a temporary workaround for accidentally making something immutable before we work out redact/immutable (i.e. I accidentally made immutable a workflow with a password).